lib/cloudex/cloudinary_api.ex

```diff
@@ -102,38 +104,39 @@ defmodule Cloudex.CloudinaryApi do
 
     body = {:multipart, [{:file, file_path} | options]}
 
-    post(body, file_path, opts)
+    post(body, file_path, opts, recv_timeout)
   end
 
   @spec extract_cloudinary_opts(map) :: map
   defp extract_cloudinary_opts(opts) do
     Map.delete(opts, :resource_type)
   end
 
-  @spec upload_url(String.t(), map) ::
+  @spec upload_url(String.t(), map, integer()) ::
           {:ok, %Cloudex.UploadedImage{}} | {:ok, %Cloudex.UploadedVideo{}} | {:error, any}
-  defp upload_url(url, opts) do
+  defp upload_url(url, opts, recv_timeout) do
     opts
     |> Map.merge(%{file: url})
     |> prepare_opts
     |> sign
     |> URI.encode_query()
-    |> post(url, opts)
+    |> post(url, opts, recv_timeout)
   end
 
-  defp credentials do
+  defp credentials(recv_timeout) do
     [
+      recv_timeout: recv_timeout,
       hackney: [
         basic_auth: {Cloudex.Settings.get(:api_key), Cloudex.Settings.get(:secret)}
       ]
     ]
   end
 
-  @spec delete_file(bitstring, map) ::
+  @spec delete_file(bitstring, map, integer()) ::
           {:ok, HTTPoison.Response.t() | HTTPoison.AsyncResponse.t()}
           | {:error, HTTPoison.Error.t()}
-  defp delete_file(item, opts) do
-    HTTPoison.delete(delete_url_for(opts, item), @cloudinary_headers, credentials())
+  defp delete_file(item, opts, recv_timeout) do
+    HTTPoison.delete(delete_url_for(opts, item), @cloudinary_headers, credentials(recv_timeout))
```

Comment:
I think it might be better to leave the `credentials` function unchanged, and do ```credentials() ++ [recv_timeout: recv_timeout```

or if we go the keyword list route ```credentials() ++ poison_opts``` where poison opts would be something like [recv_timeout: 60_0000, max_redirect: 3, proxy_auth: {user, "password"}]